app/src/renderer/js/main.js

```diff
@@ -297,14 +333,16 @@ document.addEventListener('DOMContentLoaded', () => {
       onInvalid: shake
     });
 
-    if (linkBtn.classList.contains('active')) {
-      const tmp = aspectRatio.resize(...aspectRatioBaseValues, undefined, this.value);
-      if (tmp[0]) {
-        lastValidInputWidth = tmp[0];
-        inputWidth.value = tmp[0];
-      }
+    height = this.value;
+
+    if (lockedRatio) {
+      width = (selectedRatio.split(':')[0] / selectedRatio.split(':')[1]) * this.value;
+      inputWidth.value = Math.round(width);
+      return;
     }
```

Comment:
This logic is almost the same as on line 309 and 379. Could be DRY'ed up.

Reply:
I'm having a slight bit of trouble with this one. 
It would be easy to break this logic into a function but there's another part of the code that has _very_ similar logic to the parts we want to DRY up.
(for reference, here's the logic https://github.com/wulkano/kap/pull/251/files#diff-a9a03e39450b8c7cd627d10635075959R338)
I'm wondering, because we want to create a function for working out the `width` via the selected ratio, do we want another function for the height (because it wouldn't make sense to keep that logic isolated whilst we break out the former), or is there a way we could create one function for the two.

What would be best here?

Reply:
Yeah, you're right. Not that easy. Really, all this would have been much nicer if the DOM was automatically synced from the model instead of us having to manually update it in all the places, like with React or Vue.

Would at least be a little bit nicer if all the ratio logic was encapsulated in a class. That would reduce the need for global state and make it easier to reuse some of the logic. Then we would store the ratio as an object instead of string, and just have a property that renders our internal `{width: 16, height: 9}` to `"16:9"`.

`dimensions.height = (dimensions.ratio.split(':')[1] / dimensions.ratio.split(':')[0]) * this.value;` internally could simply be `this.height = (this.ratio.height / this.ratio.width) * this.value;`